keyvi/include/keyvi/dictionary/dictionary_merger.h

```diff
@@ -63,9 +65,11 @@ struct MergeStats {
   size_t updated_keys_ = 0;
 };
 
-template <class PersistenceT, class ValueStoreT = fsa::internal::NullValueStore>
+template <int DictionaryType = fsa::internal::value_store_t::KEY_ONLY,
+          typename ValueStoreMergeT = typename fsa::internal::Dict<DictionaryType>::value_store_merge_t,
+          typename ValueStoreAppendMergeT = typename fsa::internal::Dict<DictionaryType>::value_store_append_merge_t>
```

Comment:
Do we want to have this type deduction as a default behavior which is possible to override, or it should be enforced and be the only one ?